### DISCUSSION TITLE
Making left sidebar have dropdowns for sub TOCs

### DISCUSSION THREAD

**Discussion by npetersen2** [pradyunsg/furo#318] [Questions] [answered]
Hi, I am trying to replicate the sidebar on the main furo website:

![image](https://user-images.githubusercontent.com/20168990/143322898-6fc828b0-5c56-4743-bcf5-a5075e796825.png)

Where the sidebar is collapsed by default and you can open it up to browse the sub-pages. Very nice.

However, I cannot figure out how to do this on my site...

- I have a top level page: `firmware/arch/index`
- With three children: `firmware/arch/subpage1`, `firmware/arch/subpage2`, `firmware/arch/subpage3`

In my main TOC in my top level, I link ONLY to `firmware/arch/index`. Then, inside of `firmware/arch/index`, I create a TOC for the sub pages. Both TOC are marked as hidden.

However, when the site renders, the subpages are not in a drop down! They are just shown at the same level as the `firmware/arch/index` page.

**How do I get them to appear as children of the top page?**

**Comment by pradyunsg** [comment]
Getting the sidebar to behave like this (with dropdowns, and dealing with complexity of having a collapse button) is... non-trivial with Sphinx. You can get indentation by having a `margin-left` set only on nested `ul`s.

https://github.com/pradyunsg/furo/blob/0c4018fa730d12dc2144745b75c33b922b7cf7d4/src/furo/navigation.py has most of the HTML adaptation necessary, to get things to work correctly.

https://github.com/pradyunsg/furo/blob/0c4018fa730d12dc2144745b75c33b922b7cf7d4/src/furo/assets/styles/components/_sidebar.sass has the relevant stylesheet, for getting the sidebar to look right.

**Reply by npetersen2** [reply]
Thanks, @pradyunsg for the reply!!

I am a bit new to Sphinx and theming. Are the files you linked already in the `furo` theme I have installed? as in, I do not need to do anything and the sidebar should be able to behave like the main `furo` site? Or, do I need to alter something to get my installed `furo` theme to work like the example site?

**Reply by pradyunsg** [reply]
Ah, I think I misunderstood what you were asking.

**Comment by pradyunsg** [comment, marked as answer]
Are you trying to get sidebar dropdowns in a Furo-based documentation site?

Here's a small example of how to do that:

index.rst

```
Home
====

.. toctree::
   page
   dropdown/index
```

page.rst
```
Page
====
```

dropdown/index.rst
```
Dropdown
========

.. toctree::
   inner
```

dropdown/inner.rst

```
Inner
====
```

conf.py
```
html_theme = "furo"
```

This will generate:

<img width="571" alt="Screenshot 2021-11-26 at 17 22 59" src="https://user-images.githubusercontent.com/3275593/143614801-1de8186a-5455-4443-9b6a-528cc6109bef.png">

**Reply by npetersen2** [reply]
@pradyunsg thanks for the example.

After a lot of playing, I finally figured out why mine wasn't working. I think this is a bug somewhere.

To reproduce: Simply swap the ordering of the `toctree` declaration in the dropdown rst file:

`dropdown/index.rst` becomes:

```
.. toctree::
   inner

Dropdown
========
```

Now, the dropdown doesn't render at all. Everything is just flat in the side bar.

![image](https://user-images.githubusercontent.com/20168990/143620330-17631449-0032-4dde-b71b-fadb0d99d0c2.png)

From user side, easy to fix this by always putting the `toctree` declaration at the bottom of the file.

**Reply by pradyunsg** [reply]
Looks like a bug in either `docutils` or `sphinx`.
